

{{{id=1|
_=var('al');A = matrix([[cos(al), sin(al)],[sin(al),-cos(al)]])
[ m.full_simplify() for m in  A.eigenvalues()]
///
[-1, 1]
}}}

{{{id=11|
A.eigenvectors_right()
///
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "_sage_input_32.py", line 10, in <module>
    exec compile(u'open("___code___.py","w").write("# -*- coding: utf-8 -*-\\n" + _support_.preparse_worksheet_cell(base64.b64decode("QS5laWdlbnZlY3RvcnNfcmlnaHQoKQ=="),globals())+"\\n"); execfile(os.path.abspath("___code___.py"))' + '\n', '', 'single')
  File "", line 1, in <module>
    
  File "/tmp/user/2001/tmp9IpZgN/___code___.py", line 2, in <module>
    exec compile(u'A.eigenvectors_right()' + '\n', '', 'single')
  File "", line 1, in <module>
    
  File "matrix2.pyx", line 4101, in sage.matrix.matrix2.Matrix.eigenvectors_right (sage/matrix/matrix2.c:23622)
TypeError: eigenvectors_left() takes no keyword arguments
}}}

{{{id=3|
E = identity_matrix(2)
p = (A-x*E).det(); p
///
(x - cos(al))*(x + cos(al)) - sin(al)^2
}}}

{{{id=12|
A.charpoly()
///
x^2 - sin(al)^2 - cos(al)^2
}}}

{{{id=4|
[m.full_simplify() for m in solve(p==0,x)]
///
[x == -1, x == 1]
}}}

{{{id=5|
A = matrix([[1,2,3],[4,5,6],[7,8,9]])
b1 = vector([0,0,0])
b2 = vector([1,0,0])
b3 = A*b2
print b3
print A\b1
print A\b3
///
(1, 4, 7)
(0, 0, 0)
(1, 0, 0)
}}}

{{{id=7|
A\b2
///
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "_sage_input_25.py", line 10, in <module>
    exec compile(u'open("___code___.py","w").write("# -*- coding: utf-8 -*-\\n" + _support_.preparse_worksheet_cell(base64.b64decode("QVxiMg=="),globals())+"\\n"); execfile(os.path.abspath("___code___.py"))' + '\n', '', 'single')
  File "", line 1, in <module>
    
  File "/tmp/user/2001/tmp_7V1aG/___code___.py", line 2, in <module>
    exec compile(u'A * BackslashOperator() * b2' + '\n', '', 'single')
  File "", line 1, in <module>
    
  File "/usr/local/sage/local/lib/python2.6/site-packages/sage/misc/preparser.py", line 1360, in __mul__
    return self.left._backslash_(right)
  File "matrix2.pyx", line 73, in sage.matrix.matrix2.Matrix._backslash_ (sage/matrix/matrix2.c:2789)
  File "matrix2.pyx", line 302, in sage.matrix.matrix2.Matrix.solve_right (sage/matrix/matrix2.c:3715)
  File "matrix2.pyx", line 422, in sage.matrix.matrix2.Matrix._solve_right_general (sage/matrix/matrix2.c:4488)
ValueError: matrix equation has no solutions
}}}

{{{id=8|

///
}}}